sidecar/src/service/tracing/trace_flusher.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use super::TraceSendData;
+use crate::agent_remote_config::AgentRemoteConfigWriter;
+use datadog_ipc::platform::NamedShmHandle;
+use datadog_trace_utils::trace_utils;
+use datadog_trace_utils::trace_utils::SendData;
+use ddcommon::Endpoint;
+use futures::future::join_all;
+use manual_future::ManualFuture;
+use serde::{Deserialize, Serialize};
+use std::collections::hash_map::Entry;
+use std::collections::{BTreeMap, HashMap};
+use std::iter::zip;
+use std::ops::DerefMut;
+use std::sync::atomic::{AtomicU32, AtomicU64, Ordering};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+use tokio::select;
+use tokio::task::{JoinError, JoinHandle};
+use tracing::{debug, error, info};
+
+/// `TraceFlusherStats` holds stats of the trace flusher like the count of allocated shared memory
+/// for agent config, agent config writers, last used entries in agent configs, and the size of send
+/// data.
+#[derive(Serialize, Deserialize)]
+pub struct TraceFlusherStats {
```

Comment:
Not sure if "stats" belong in their own namespace, or under telemetry, or close to the functionality they are "stat'ing" so put them near the functionality for now.

Reply:
I think they should belong near the functionality as they are explicitly meant to describe the state of the functionality, primarily as a debugging mechanism.

Reply:
@bwoebi That's a fair point. Near the functionality works for me.